### DISCUSSION TITLE
Few questions about tags and tagging support

### DISCUSSION THREAD

**Discussion by Qronikarz** [kanriapp/kanri#740] [General]
Hello.
Both Q&A and Show and Tell categories are applicable to my topic here, so I chose general. Feel free to recategorize if needed.

Because Kanri offers ability of tagging cards I have included it in my collection of programs that focus on such ability.
And because the discussions are open and Kanri entry is nearly complete I decided to reach out here.

I maintain https://github.com/Qronikarz/TagsResearch which as I wrote before is a collection of programs that offer tagging capability of various items/files/notes etc.

I have few questions that I wasn't able to find answers to.
Here's Kanri entry https://github.com/Qronikarz/TagsResearch/blob/main/components/kanri.md

1. Do you follow any tag specifications or are they custom made? I've seen JSON export, but I don't know how they are stored inside of the program.
2. Are there any required dependencies? For example if you have a clean Windows install do you need to install something. Something that could also be valid is a browser or a Windows 10 instead of 7
3. I did not find any accessibility options, but did not want to mark it yet. Something like changing fonts, text reader support, semantic html, alt texts etc. There are custom themes, but it would at least need high contrast to classify it as accessibility.
4. Is there any link to documentation about tags? I know they are very simple in Kanri, but I need to know how to mark it.
5. Are there any symbols that are forbidden while naming the tags? For example there are tagging programs that use file names on Windows which means characters such as / \ ? * etc can't be used in tags.
6. When clicking on a card it opens it in a lightbox. Do you agree with that naming or is it something else?

Those six questions are ones that are missing from Kanri entry. Besides that I think everything is classified correctly. You are of course welcome to check it yourself now and after I complete the whole entry, then you could fact check it. I have a special checkmark for entries fact checked by actual developers of programs. So far only my own project has it, but I didn't reach out to all developers.

**Comment by trobonox** [comment]
Hi, thanks for reaching out. Here are the answers to your questions:
1. The tags are custom-made, there is no specification I follow.
2. Windows 10 is the minimum required version and you might need to install WebView2 on older or stripped down versions.
3. You are correct, there are no accesibility options unfortunately. This is something I was not able to look into yet and especially a very complex layout like the Kanban board would probably need a lot of work to be made fully accessible. I might not have the expertise in this area, but as always I welcome any contributions.
4. There is no documentation of the Tags feature at the moment. I have considered writing a small manual with all app features, but the documentation is a heavy work in progress at the moment.
5. I don't do any validation in particular but there might be special characters that depending on the operating system create some issues. Because I rely on frameworks like Vue and Tauri, I can't answer this question completely.
6. I like calling it a Modal because that's the more commonly used name in UI design, but lightbox seems to be an equivalent term so it should be correct.

I have checked the rest of the entry and it looks good to me.

**Reply by Qronikarz** [reply]
Alright. So just a quick update on the entry for you to not require reading it fully again. Or you can check a commit.

> 
> * Marked the entry as complete and fact checked by you today as of v0.8.0 with link to above post.
> * Tag specification - X
> * Supported OS added "10 and above" to Windows
> * Supported filesystems - all
> * required dependencies - Webview 2
> * accessibility support - X
> * tag support links - X
> * type of tagging - custom code
> * not allowed naming symbols - X with a footnote explaining the point 5.
> * changed lightbox to modal

I think that's all the changes.
There's also contributors list both in Kanri and in the main readme.md where people who contributed can appear, but that's only if you want to. (you don't need to make any pull requests, I can just write you in both places there or if you want in only one of them)

ad 3. Understandable. I also do not have it currently because I thought nobody at all would have wanted to use it let alone people who need more polished and specialized programs, but during TagsResearch I unfortunately found out that there aren't that many programs with such capabilities. I am rewriting and remaking my whole project to have semantic HTML and alt text for images + I will try to test it with screen reader to see if it's navigable by it. So can't currently offer much help here, other than those 2 things.

ad 4. There's no need like I said because this category was mainly created for programs that didn't have tags at the beginning and the support was added later or in case of default Linux file browsers still in the "feature request" or "planning" stage.

That would be all. Big thank you for responding so quickly and I guess enjoy being the first to have a complete and fact checked entry in TagsResearch.
Good luck in finishing the v1 of Kanri. I definitely look forward to it from my quick yesterday testing since I love portable offline programs.

**Reply by gmalone** [reply]
I have to concur that TAGS need a bit more conformity with norms.  They work, but could use more finesse.  That said, great job on Kanri.
